eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageCodecs.scala

```diff
@@ -333,36 +328,26 @@ object LightningMessageCodecs {
   val channelUpdateChecksumCodec =
     ("chainHash" | bytes32) ::
       ("shortChannelId" | shortchannelid) ::
-      (messageFlagsCodec >>:~ { messageFlags =>
-        channelFlagsCodec ::
-          ("cltvExpiryDelta" | cltvExpiryDelta) ::
-          ("htlcMinimumMsat" | millisatoshi) ::
-          ("feeBaseMsat" | millisatoshi32) ::
-          ("feeProportionalMillionths" | uint32) ::
-          ("htlcMaximumMsat" | conditional(messageFlags.optionChannelHtlcMax, millisatoshi))
-      }).derive[MessageFlags].from {
-        // The purpose of this is to tell scodec how to derive the message flags from the data, so we can remove that field
-        // from the codec definition and the case class, making it purely a serialization detail.
-        // see: https://github.com/scodec/scodec/blob/series/1.11.x/unitTests/src/test/scala/scodec/examples/ProductsExample.scala#L108-L127
-        case _ :: _ :: _ :: _ :: _ :: htlcMaximumMsat_opt :: HNil => MessageFlags(optionChannelHtlcMax = htlcMaximumMsat_opt.isDefined)
-      }
+      ("messageFlags" | constant(hex"01")) ::
```

Comment:
This drops the following `Unit` codec and removes the need for `:: () ::` when applying the codec

```suggestion
      ("messageFlags" | constant(hex"01")) :~>:
```

Reply:
Neat thanks, done in https://github.com/ACINQ/eclair/pull/2361/commits/06b4ad952e69524ffedeec2adaad516f5e6e75ae